### PR TITLE
Fix #17065 make the generic method usable without token

### PR DESCRIPTION
**Pull request by ikramBej**
https://issues.rudder.io/issues/17065

### CODE REVIEW

File: consul/share/plugins/consul/30_generic_methods/variable_from_consul.cf [hunk 1]
```diff
@@ -39,19 +39,21 @@ bundle agent variable_from_consul(variable_prefix, variable_name, path)
...
       "class_prefix" string => canonify(join("_", "promisers"));
       "args" slist => {"${variable_prefix}", "${variable_name}", "${path}"};
+    
+  pass1::
+      "${variable_prefix}.${variable_name}" string => execresult('/usr/bin/curl -s ${consul.auth} http://${consul.config[server_addr]}/v1/kv/${path}?raw', "noshell");
 
-    pass1::
-            "${variable_prefix}.${variable_name}" string => execresult('/usr/bin/curl -s -H "X-Consul-Token: ${consul.config[token]}" http://${consul.config[server_addr]}/v1/kv/${path}?raw', "noshell");
-
+  reports:
```
Comment (amousset) [inline consul/share/plugins/consul/30_generic_methods/variable_from_consul.cf:46]: Debug reports I guess,should not be here

File: consul/share/plugins/consul/30_generic_methods/variable_from_consul.cf [hunk 2]
```diff
@@ -39,19 +39,21 @@ bundle agent variable_from_consul(variable_prefix, variable_name, path)
       "promisers" slist => { @{this.callers_promisers}, cf_null }, policy => "ifdefined";
       "class_prefix" string => canonify(join("_", "promisers"));
       "args" slist => {"${variable_prefix}", "${variable_name}", "${path}"};
+    
+  pass1::
```
Comment (amousset) [inline consul/share/plugins/consul/30_generic_methods/variable_from_consul.cf:43]: indentation should not change here

File: consul/share/plugins/consul/30_generic_methods/variable_from_consul.cf [hunk 3]
```diff
@@ -39,19 +39,18 @@ bundle agent variable_from_consul(variable_prefix, variable_name, path)
       "promisers" slist => { @{this.callers_promisers}, cf_null }, policy => "ifdefined";
       "class_prefix" string => canonify(join("_", "promisers"));
       "args" slist => {"${variable_prefix}", "${variable_name}", "${path}"};
-
+    
     pass1::
-            "${variable_prefix}.${variable_name}" string => execresult('/usr/bin/curl -s -H "X-Consul-Token: ${consul.config[token]}" http://${consul.config[server_addr]}/v1/kv/${path}?raw', "noshell");
```
Comment (amousset) [inline consul/share/plugins/consul/30_generic_methods/variable_from_consul.cf:44]: indentation to fix here too

Comment (ikramBej) [reply consul/share/plugins/consul/30_generic_methods/variable_from_consul.cf:44]: I think i fixed it in the last commit

Comment (amousset) [reply consul/share/plugins/consul/30_generic_methods/variable_from_consul.cf:44]: It seems you only fixed the line before but not the line with the command itself.

**Pull request merged**